clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -619,11 +625,13 @@ public void beginTransaction() throws ProducerFencedException {
      * and should also not commit offsets manually (via {@link KafkaConsumer#commitSync(Map) sync} or
      * {@link KafkaConsumer#commitAsync(Map, OffsetCommitCallback) async} commits).
      *
-     * @throws IllegalStateException if no transactional.id has been configured or no transaction has been started
+     * @throws IllegalStateException if no transactional.id has been configured, no transaction has been started,
```

Comment:
More of a meta comment, but after this  patch, it would be possible for users to use ephemeral transactional Ids since we can rely on the group coordinator fencing. One of the potential follow-ups is to figure out how to make this work from a security perspective. For example, we could piggyback on Group write permission to enforce access.

Reply:
So the producer.id generation is guaranteed to be unique by Zk? If that's the case, I also agree that we should cleanup txn.id security logic. Got 2 tickets to track: https://issues.apache.org/jira/browse/KAFKA-9453
https://issues.apache.org/jira/browse/KAFKA-9454